packages/scene-composer/src/reducers/materialReducer.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { useReducer } from 'react';
+import { act, cleanup, renderHook } from '@testing-library/react-hooks';
+import { Mesh, MeshBasicMaterial, Color } from 'three';
+
+import {
+  materialReducer,
+  initialMaterialMaps,
+  backUpOriginalMaterial,
+  addMaterial,
+  removeMaterial,
+} from './materialReducer';
+
+describe('materialReducer', () => {
+  const transformedColor = new Color(0x000000);
+  const secondTransformColor = new Color(0xff0000);
+  const originalColor = new Color(0x00ff00);
+
+  it('should backup a material', () => {
+    const [materialMaps, dispatch] = renderHook(() => useReducer(materialReducer, initialMaterialMaps)).result.current;
+
+    const mesh = new Mesh(undefined, new MeshBasicMaterial({ color: originalColor }));
+
+    act(() => {
+      backUpOriginalMaterial(mesh, materialMaps, dispatch);
+    });
+    const backUpColor = materialMaps.original[mesh.uuid].color;
+    expect(backUpColor.getHex()).toBe(originalColor.getHex());
+
+    cleanup();
```

Comment:
act is because we're calling dispatch under the hood as it's a reducer test and we get warnings if we don't.  Cleanup is unnecessary, I think I read it as a best practice but https://kentcdodds.com/blog/common-mistakes-with-react-testing-library says it's usually not necessary now.  Runs fine without when I tried it locally.